enterprise/server/remote_execution/containers/sandbox/BUILD

```diff
@@ -24,12 +24,21 @@ go_test(
     tags = [
         "no-sandbox",  # sandbox-exec is not compatible with Bazel's sandbox environment
     ],
-    target_compatible_with = ["@platforms//os:macos"],
-    deps = [
-        ":sandbox",
-        "//enterprise/server/remote_execution/container",
-        "//proto:remote_execution_go_proto",
-        "//server/testutil/testfs",
-        "@com_github_stretchr_testify//assert",
-    ],
+    deps = select({
+        "@io_bazel_rules_go//go/platform:darwin": [
+            ":sandbox",
+            "//enterprise/server/remote_execution/container",
+            "//proto:remote_execution_go_proto",
+            "//server/testutil/testfs",
+            "@com_github_stretchr_testify//assert",
+        ],
+        "@io_bazel_rules_go//go/platform:ios": [
+            ":sandbox",
+            "//enterprise/server/remote_execution/container",
+            "//proto:remote_execution_go_proto",
+            "//server/testutil/testfs",
+            "@com_github_stretchr_testify//assert",
+        ],
+        "//conditions:default": [],
```

Comment:
Review note (to my future self): When the test is built for Linux, rules_go will inject an `empty.go` file into both internal and external test binary, effectively creating 2 no-ops tests on non-darwin platforms.

Bazel would still display that the test was executed on Linux, but it does not run any actual Go tests.